src/shared/bootspec.c

```diff
@@ -1482,25 +1477,25 @@ static int print_cmdline(
                 if (!t2)
                         return log_oom();
 
-                printf("  ukiCmdline: %s\n", t2);
-                final_cmdline = TAKE_PTR(t2);
+                printf("      options: %s\n", t2);
+                combined_cmdline = TAKE_PTR(t2);
         }
 
-        FOREACH_ARRAY(addon, global_arr->items, global_arr->count) {
-                print_addon(addon, "globalAddon");
-                if (!strextend(&final_cmdline, " ", addon->cmdline))
+        FOREACH_ARRAY(addon, global_arr->items, global_arr->n_items) {
+                print_addon(addon, "globaladdon");
+                if (!strextend(&combined_cmdline, " ", addon->cmdline))
                         return log_oom();
         }
 
-        FOREACH_ARRAY(addon, e->local_addons.items, e->local_addons.count) {
+        FOREACH_ARRAY(addon, e->local_addons.items, e->local_addons.n_items) {
                 /* Add space at the beginning of addon_str to align it correctly */
-                print_addon(addon, " localAddon");
-                if (!strextend(&final_cmdline, " ", addon->cmdline))
+                print_addon(addon, " localaddon");
+                if (!strextend(&combined_cmdline, " ", addon->cmdline))
                         return log_oom();
         }
 
-        if (final_cmdline)
-                printf(" finalCmdline: %s\n", final_cmdline);
+        if (combined_cmdline && (e->local_addons.n_items + global_arr->n_items) > 0)
```

Comment:
maybe just do `!strq_ptr(combined_cmdline, t)` here. and then make sure that `t` above is defined at the right scope...

Reply:
I don't understand what strq_ptr makes, and honestly also why comparing it with t. At most with t2 no? and remove the TAKE_PTR from it